src/containers/EntryPage.js

```diff
@@ -59,11 +59,14 @@ class EntryPage extends React.Component {
       loadEntry(collection, slug);
     }
 
-    const unblock = history.block((location) => {
+    this.blocker = () => {
       if (this.props.entryDraft.get('hasChanged')) {
         return "Are you sure you want to leave this page?";
       }
-    });
+    }
+
+    const unblock = history.block(this.blocker);
+    window.onbeforeunload = window.onbeforeunload || this.blocker;
```

Comment:
Just wonder if using `addEventListener` and `removeEventListener` would be better here.

Reply:
`window.addEventListener(this.blocker)`

Reply:
:+1: